src/lib/crypto/kimchi_bindings/stubs/src/caml/caml_pointer.rs

```diff
@@ -1,7 +1,7 @@
 macro_rules! impl_caml_pointer {
     ($name: ident => $typ: ty) => {
         #[derive(std::fmt::Debug, Clone, ::ocaml_gen::CustomType)]
-        pub struct $name(pub ::std::rc::Rc<$typ>);
+        pub struct $name(pub ::std::rc::Rc<std::cell::UnsafeCell<$typ>>);
```

Comment:
Note as it is important for the Caml binding: [UnsafeCell](https://doc.rust-lang.org/src/core/cell.rs.html#2071) is implemented as:
```
#[lang = "unsafe_cell"]
#[stable(feature = "rust1", since = "1.0.0")]
#[repr(transparent)]
#[rustc_pub_transparent]
pub struct UnsafeCell<T: ?Sized> {
    value: T,
}
```

Reply:
I guess we're adding a new indirection? We might lose some performances?

Reply:
> Note as it is important for the Caml binding: UnsafeCell is implemented as:

The nightly tests were passing with this change, does this mean that that the ocaml bindings lib was able to resolve this on it's own or is it just coincidence because maybe the runtime rep is the same

Reply:
> I guess we're adding a new indirection? We might lose some performances?

My intuition is that this `UnsafeCell` [should be optimized away](https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html#advantages)